cedar-wasm/src/utils.rs

```diff
@@ -0,0 +1,38 @@
+use cedar_policy::ffi::{Schema, Template};
+use serde::{Deserialize, Serialize};
+use tsify::Tsify;
+use wasm_bindgen::prelude::*;
+
+#[derive(Tsify, Debug, Serialize, Deserialize)]
+#[serde(tag = "type")]
+#[serde(rename_all = "camelCase")]
+#[tsify(into_wasm_abi, from_wasm_abi)]
+/// struct that defines the result of getting valid request environments
+pub enum GetValidRequestEnvsResult {
+    /// represents a successful call to [`get_valid_request_envs()`]
+    Success {
+        principals: Vec<String>,
+        actions: Vec<String>,
+        resources: Vec<String>,
+    },
+    /// represents a failed call to [`get_valid_request_envs()`]
+    Error {
```

Comment:
For consistency with the other Wasm/FFI functions
```suggestion
    Failure {
```